packages/mattermost-redux/src/actions/files.ts

```diff
@@ -17,6 +18,29 @@ export function receivedFiles(files: Map<string, FileSearchResultItem>): Action
     };
 }
 
+export function getMissingFilesByPosts(posts: Post[]) {
+    return (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const {files} = getState().entities.files;
+        const postIds = posts.reduce((curr: Array<Post['id']>, post: Post) => {
+            const {file_ids: fileIds} = post;
+            if (!fileIds || fileIds.every((id) => files[id])) {
+                return curr;
+            }
+            curr.push(post.id);
+
+            return curr;
+        }, []);
+
+        const promises: Array<Promise<any>> = [];
+
+        for (const id of postIds) {
```

Comment:
Wonder if we should worry about this list to be big. It would dispatch many times "at the same time" the action, which I guess is making calls to the server.

Reply:
@larkox although this should be a small subset, if any, of the 25 threads per page 
(threads that have files in the root post, and those files are not yet known, fyi, files are loaded when visiting the channel),
you are right, some kind of batching is appropriate here.

Reply:
@hmhealey when collecting threads we are not collecting files for posts, adding that there might burden an already heavy operation, thus deciding to do the fetching separately.

Added a ticket for batching https://mattermost.atlassian.net/browse/MM-43297